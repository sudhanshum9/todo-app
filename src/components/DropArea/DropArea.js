import { useState } from "react";


function DropArea({ onDropTask }) {
  const [showDrop, setShowDrop] = useState(false);

  return (
    <section
      aria-label="drop-area-section"
      onDragEnter={() => setShowDrop(true)}
      onDragLeave={() => setShowDrop(false)}
      onDrop={() => {
        onDropTask();
        setShowDrop(false);
      }}
      onDragOver={(e) => e.preventDefault()}
      style={{
        height:showDrop? '100px': '10px',
        backgroundColor: showDrop ? '#f0f0f0' : '#fff',  // Change background color when dragging
        border: showDrop ? '1px dashed #dcdcdc': 0,  // Change border when dragging
        borderRadius: '10px',
        padding: '12px',
        margin: showDrop ? '12px 0': 0,
        textAlign: 'center',
        opacity: showDrop ? 1 : 0,  // Change opacity when dragging
        transition: 'all 0.2s ease-in-out',
      }}
    >
      Drop Area
    </section>
  );
}

export default DropArea;
