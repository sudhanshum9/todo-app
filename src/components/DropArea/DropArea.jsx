import { useState } from "react";


function DropArea({ onDropTask }) {
  const [showDrop, setShowDrop] = useState(false);
  return (
    <section
      aria-label="drop-area-section"
      onDragEnter={() => setShowDrop(true)}
      onDragLeave={() => setShowDrop(false)}
      onDrop={() => {
        onDropTask();
        setShowDrop(false);
      }}
      onDragOver={(e) => e.preventDefault()}
      style={{
        width: '100%',
        height: '100px',
        backgroundColor: showDrop ? '#f0f0f0' : '#fff',  // Change background color when dragging
        border: showDrop ? '2px solid #000' : '1px dashed #dcdcdc',  // Change border when dragging
        borderRadius: '10px',
        padding: '15px',
        marginBottom: '15px',
        opacity: showDrop ? 1 : 0,  // Change opacity when dragging
        transition: 'all 0.2s ease-in-out',
        cursor: showDrop ? 'pointer' : 'default',  // Change cursor style when dragging
      }}
    >
      Drop Area
    </section>
  );
}

export default DropArea;
