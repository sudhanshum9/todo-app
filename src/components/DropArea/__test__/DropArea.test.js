import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import DropArea from '../DropArea';

describe('DropArea Component', () => {

  test('renders correctly and matches snapshot', () => {
    render(<DropArea onDropTask={jest.fn()} />);
    expect(screen.getByText(/Drop Area/i)).toMatchSnapshot();
  });

  test('changes style when drag enters and leaves', () => {
    render(<DropArea onDropTask={jest.fn()} />);
    const dropArea = screen.getByText(/Drop Area/i);
    
    // Initially, the opacity should be 0 and border dashed
    expect(dropArea).toHaveStyle('opacity: 0');
    expect(dropArea).toHaveStyle('border: 1px dashed #dcdcdc');

    // Simulate drag enter
    fireEvent.dragEnter(dropArea);
    expect(dropArea).toHaveStyle('opacity: 1');
    expect(dropArea).toHaveStyle('border: 2px solid #000');
    expect(dropArea).toHaveStyle('background-color: #f0f0f0');

    // Simulate drag leave
    fireEvent.dragLeave(dropArea);
    expect(dropArea).toHaveStyle('opacity: 0');
    expect(dropArea).toHaveStyle('border: 1px dashed #dcdcdc');
  });

  test('calls onDropTask when an item is dropped', () => {
    const onDropTaskMock = jest.fn();
    render(<DropArea onDropTask={onDropTaskMock} />);
    const dropArea = screen.getByText(/Drop Area/i);

    // Simulate dropping an item
    fireEvent.drop(dropArea);
    
    // Check if the onDropTask function was called
    expect(onDropTaskMock).toHaveBeenCalled();
  });

  test('renders with correct accessible attributes', () => {
    render(<DropArea onDropTask={jest.fn()} />);
    const dropArea = screen.getByLabelText(/drop-area-section/i); // Check for aria-label
    
    // Ensure the section has aria-label="Drop Area" for better accessibility
    expect(dropArea).toBeInTheDocument(); // aria-label should now make this pass
  });

});
