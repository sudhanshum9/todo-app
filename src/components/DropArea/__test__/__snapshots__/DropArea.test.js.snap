// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropArea Component renders correctly and matches snapshot 1`] = `
<section
  aria-label="drop-area-section"
  style="width: 100%; height: 100px; background-color: rgb(255, 255, 255); border: 1px dashed #dcdcdc; border-radius: 10px; padding: 15px; margin-bottom: 15px; opacity: 0; transition: all 0.2s ease-in-out; cursor: default;"
>
  Drop Area
</section>
`;
